### PR TITLE
Bug 1977383: [release-4.7] Ensure service ca configmap is created in all namespaces

### PR DESCRIPTION
**Pull request by marun**
This is a 4.7 followup to https://github.com/openshift/kubernetes/pull/714 to ensure upgrades from 4.7 to 4.8 are not disrupted by the lack of service ca configmaps before BoundServiceAccountTokenVolume is enabled by the 4.8 kube-apiserver. 

---
A new controller is added to create a configmap per namespace that is annotated for service ca injection. The controller is derived from the controller that creates configmaps for the root ca.

This new controller is being added in 4.7 to ensure that the service ca configmaps will be present in all namespaces in the cluster before an upgrade to 4.8 is attempted. An upgrade to 4.8 will enable the BoundServiceAccountTokenVolume feature and all new pods will expect the configmaps to already be present.

A similar strategy was pursued upstream, in that publication of root ca configmaps was added in a release previous to enablement of BoundServiceAccountTokenVolume.

/cc @s-urbaniak @stlaz @sttts @soltysh

### CODE REVIEW

**Comment by marun** [comment]
Marking WIP until  openshift/origin#26283 merges.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Comment by marun** [comment]
[this e2e-gcp result](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/26284/pull-ci-openshift-origin-release-4.7-e2e-gcp/1409895553227034624) is very strange. It failed on the test that *should* have been disabled by the merge of https://github.com/openshift/origin/pull/26283, and the output doesn't make sense since it appears to include log lines that would only be added once https://github.com/openshift/origin/pull/26284 merges. Hopefully a retest will reflect the current state of origin@release-4.7.

Also, updated the resource quota configmap conformance test to reflect the addition of the openshift-service-ca.crt configmap in every namespace. [This result](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_kubernetes/834/pull-ci-openshift-kubernetes-release-4.7-k8s-e2e-conformance-aws/1410046856053919744) indicated that the test was not accounting for the new configmap. As of 1.21 (4.8 and above), the test in question sets the test resource quota for configmaps to 10, so this fix was not required for https://github.com/openshift/kubernetes/pull/714.

File: test/e2e/apimachinery/resource_quota.go
Comment (marun) [inline test/e2e/apimachinery/resource_quota.go:1481]: Double-entry commenting:

This change is required for 4.7 to ensure the test can pass with the addition of the new service ca configmap. The test code was updated in 1.21 (4.8 and above) to set the configmap quota to 10, so this change was not required for the addition of service ca configmap publishing to pass conformance in the 4.8 PR.

**Comment by sdodson** [comment]
Test changes have finally made it into an accepted CI payload.
/retest

**Comment by marun** [comment]
Rebased to pick up #839

**Comment by marun** [comment]
The watch consistency test failure is intermittent without this PR but consistent with it. https://github.com/openshift/kubernetes/pull/843 backports the already-merged upstream fix and will need to be vendored via https://github.com/openshift/origin/pull/26298 before this PR can safely merge.

**Comment by sdodson** [comment]
/retest
https://github.com/openshift/origin/pull/26303 has merged

File: openshift-kube-controller-manager/servicecacertpublisher/publisher.go
Comment (deads2k) [inline openshift-kube-controller-manager/servicecacertpublisher/publisher.go:168]: Is this injecting the same content contained in token secret's service-ca.crt?  I remember seeing a bug that the content is different.

I want to see pods start properly during upgrades, but if we know that we're going to change the included content, it seems like we want to place the correct content in there to begin with.

Comment (marun) [reply openshift-kube-controller-manager/servicecacertpublisher/publisher.go:168]: As per the thread on slack, my understanding was that we do not want to maintain token secret parity. Holding the PR until this discussion is resolved one way or another.

**Comment by marun** [comment]
metal ipi is definitely unrelated and likely points to a broken job.

**Comment by deads2k** [comment]
/lgtm
/hold cancel

This doesn't harm 4.7.z and if we decide to go this direction, merging early will assist.

**Comment by sdodson** [comment]
/override ci/prow/e2e-metal-ipi
Failed at infrastructure provisioning.

**Comment by deads2k** [comment]
We have to eat this change forever.

**Comment by marun** [comment]
/retest

Not sure why we're still gating on the cmd job? iirc it's not known to pass reliably against master.

**Comment by sdodson** [comment]
/override ci/prow/e2e-aws-downgrade
Flakey since July 1
/override ci/prow/e2e-aws-serial
No chance this is related and has passed on previous runs of same commit.
/override ci/prow/e2e-aws-jenkins
Passed previously on the same commit.

**Pull request merged**
